### PR TITLE
Fred bowker

### PR DESCRIPTION
**Pull request by fred-bowker**

### CODE REVIEW

**Comment by jaesung-0o0** [comment]
Image overriding feature has been merged into main branch.
Please rebase to the latest commit of main branch.

**Comment by fred-bowker** [comment]
Hi I fixed the merge conflicts for pulling into my fork, can you approve this review so that I can merge,

**Comment by jaesung-0o0** [comment]
Hi,
Please allow me to have a write permission

File: Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            Spacer()
+            HStack {
+                Spacer()
+                Image(getImageName() , bundle: .module)
+                        .padding(.bottom, CustomTabBarController().tabBarHeight + PADDING_ABOVE_TAB)
+            }
+        }
+    }
+
+    private func getImageName() -> String {
```
Comment (jaesung-0o0) [inline Sources/ChatUI/ChatInChannel/MessageField/Giphy/View/GiphyAttributionMarkView.swift:26]:
```suggestion
    private va imageName: String {
```
it doesn't need to use method. I recommend that you use a read-only property instead.

File: Sources/ChatUI/ChatInChannel/MessageField/Location/View/LocationSelector.swift [hunk 2]
```diff
@@ -72,7 +72,9 @@ public struct LocationSelector: View {
                 // Dimiss button & Search bar
                 HStack {
                     Button(action: dismiss) {
+
```
Comment (jaesung-0o0) [inline Sources/ChatUI/ChatInChannel/MessageField/Location/View/LocationSelector.swift:75]: I found some unnecessary vertical white spaces.  Please allow me to modify by granting the writing permission

File: Sources/ChatUI/ChatInChannel/MessageField/MessageField.swift [hunk 3]
```diff
@@ -58,168 +58,189 @@ public struct MessageField: View {
...
+
                             appearance.images.camera(colorScheme).medium
+
                         }
-                        .tint(appearance.tint)
-                        .disabled(isMenuItemPresented)
-                        .frame(width: 36, height: 36)
+                                .tint(appearance.tint)
+                                .disabled(isMenuItemPresented)
+                                .frame(width: 36, height: 36)
```
Comment (jaesung-0o0) [inline Sources/ChatUI/ChatInChannel/MessageField/MessageField.swift:114]: They need to be reset by removing the indents

File: Sources/ChatUI/Essentials/GiphyConfiguration.swift [hunk 4]
```diff
@@ -7,18 +7,21 @@ public struct GiphyConfiguration {
     public let shouldLocalizeSearch: Bool
     public let mediaTypeConfig: [GiphyUISDK.GPHContentType]
     public let presentationDetents: CGFloat
+    public let showAttributionMark: Bool
```
Comment (jaesung-0o0) [inline Sources/ChatUI/Essentials/GiphyConfiguration.swift:10]:
```suggestion
    public let showsAttributionMark: Bool
```
I recommend that you name the boolean type property as 3rd person singular verb.
Also line#6
```swift
- showConfirmationScreen: Bool
+ showsConfirmationScreen: Bool
```

File: Sources/ChatUI/Essentials/GiphyConfiguration.swift [hunk 5]
```diff
@@ -7,18 +7,21 @@ public struct GiphyConfiguration {
...
 
     public init(
             dimBackground: Bool = false,
             showConfirmationScreen: Bool = false,
             shouldLocalizeSearch: Bool = false,
             mediaTypeConfig: [GiphyUISDK.GPHContentType] = [.gifs, .stickers, .recents],
-            presentationDetents: CGFloat = 0.9
+            presentationDetents: CGFloat = 0.9,
+            showAttributionMark: Bool = false
```
Comment (jaesung-0o0) [inline Sources/ChatUI/Essentials/GiphyConfiguration.swift:18]:
```suggestion
            showsAttributionMark: Bool = false
```

**Review by x-0o0 — COMMENTED** [review]
Thanks for the amazing PR again.

I left some comments regarding style guide. Also, please allow me to have write permission.

File: Sources/ChatUI/Essentials/GiphyConfiguration.swift (hunk 4 above)
Comment (fred-bowker) [reply Sources/ChatUI/Essentials/GiphyConfiguration.swift:10]: Hi I pushed this commit accidentally, it should be a separate pull request, https://support.giphy.com/hc/en-us/articles/360035158592-What-conditions-does-my-app-project-need-to-meet-in-order-to-get-a-production-API-Key-

Comment (fred-bowker) [reply Sources/ChatUI/Essentials/GiphyConfiguration.swift:10]: is there any discourse chat server or other chat way to discuss any release dates for the OSS project? thanks for creating this project

**Pull request closed without merging**